test_suite/stdlib.jsonnet

```diff
@@ -1530,4 +1529,74 @@ std.assertEqual(std.all([true, false]), false) &&
 std.assertEqual(std.all([true, true]), true) &&
 std.assertEqual(std.all([]), true) &&
 
+std.assertEqual(std.regexFullMatch(@'e', 'hello'), null) &&
+
+std.assertEqual(
+  std.regexFullMatch(@'h.*o', 'hello'),
+  {
+    string: 'hello',
+    captures: [],
+    namedCaptures: {},
+  }
+) &&
+
+std.assertEqual(
+  std.regexFullMatch(@'h(.*)o', 'hello'),
+  {
+    string: 'hello',
+    captures: ['ell'],
+    namedCaptures: {},
+  }
+) &&
+
+std.assertEqual(
+  std.regexFullMatch(@'h(?P<mid>.*)o', 'hello'),
+  {
+    string: 'hello',
+    captures: ['ell'],
+    namedCaptures: {
+      mid: 'ell',
+    },
+  }
+) &&
+
+std.assertEqual(std.regexPartialMatch(@'world', 'hello'), null) &&
+
+std.assertEqual(
+  std.regexPartialMatch(@'e', 'hello'),
+  {
+    string: 'hello',
+    captures: [],
+    namedCaptures: {},
+  }
+) &&
+
+std.assertEqual(
+  std.regexPartialMatch(@'e(.*)o', 'hello'),
+  {
+    string: 'hello',
+    captures: ['ll'],
+    namedCaptures: {},
+  }
+) &&
+
+std.assertEqual(
+  std.regexPartialMatch(@'e(?P<mid>.*)o', 'hello'),
+  {
+    string: 'hello',
+    captures: ['ll'],
+    namedCaptures: {
+      mid: 'll',
+    },
+  }
+) &&
+
+std.assertEqual(std.regexQuoteMeta(@'1.5-2.0?'), '1\\.5\\-2\\.0\\?') &&
```

Comment:
This quotes a string to be included in regexp verbatim, right?

I'm not sure if we want to offer this. Is there a very compelling use case? I would prefer to err on the side of having smaller API (easier to add than to remove).

Reply:
Given that we are already locking into RE2, reducing the API surface isn't going to make it any easier to migrate to a different regex provider.  In terms of implementation, it's a small wrapper around the underlying call.

In terms of usefulness, if you want to search for a computed something inside something else I suppose it would be invaluable.

I'm ok with it.